app/controllers/fields_controller.rb

```diff
@@ -0,0 +1,30 @@
+class FieldsController < AuthenticatedController
+  # Returns the form view given a source text
+  def form
+    @form_data = HasFields.parse_fields(params[:source])
+    @allow_dropdown = params[:allow_dropdown] == 'true'
```

Comment:
I'm trying really hard to figure out a way around this. It's scattered in so many places. I can see why, I totally get why it's here I just wish we didn't need it.

Just to prevent how we're currently spreading it through each view etc. Why not just reference it directly?

`<% @allow_dropdown && options = value.split(' | ') %>`

The partials have access to the variable it would tidy up the forced passing through local assigns.